### PR TITLE
feature/intel-mac-support

### PR DESCRIPTION
**Pull request by nezda**
WIP make get_number_of_sockets and CpuInfo.get_number_of_sockets work on intel mac

## Type of Change

feature or bug fix or documentation or validation or others  
API changed or not

## Description

detail description

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

File: neural_compressor/utils/utility.py [hunk 1]
```diff
@@ -918,6 +920,8 @@ def get_number_of_sockets() -> int:
     cmd = "lscpu | grep 'Socket(s)' | cut -d ':' -f 2"
     if sys.platform == "win32":
         cmd = 'wmic cpu get DeviceID | find /c "CPU"'
+    elif sys.platform == "darwin":
+        cmd = "sysctl -n machdep.cpu.core_count"
```
Comment (nezda) [inline neural_compressor/utils/utility.py:924]: technically i never tested this duplicate looking fn

Comment (Spycsh) [reply neural_compressor/utils/utility.py:924]: Yes I think it is just a duplicate method, but to keep this PR clean, we just keep it.

**Comment by Spycsh** [comment]
@nezda thanks for this PR. Looks good to me. Let's wait and see whether the UT pass. If there are coverage issues, please add `# pragma: no cover` to your code.

**Review by Spycsh — COMMENTED** [review]
@nezda  Please exclude code from coverage check by using `pragma: no cover` since there are no tests.

**Comment by nezda** [comment]
@Spycsh think I got that right.

**Review by Spycsh — APPROVED** [review]

**Comment by Spycsh** [comment]
Hi @nezda thanks, already approved! Hi @chensuyue , could you have a check and merge this?

**Pull request merged**